### PR TITLE
DWR-1060 Migrate Olap - added student ethnicity

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 1]
```diff
@@ -105,4 +105,47 @@ sql:
...
+        # ------------ student_ethnicity  -------------------------------------------------------------------
+        student_ethnicity:
+          sql:
+            warehouseRead: >-
+              SELECT
+                ethnicity_id,
+                student_id
+              FROM student_ethnicity
+                JOIN student ws ON ws.id = student_ethnicity.student_id
+              WHERE ? IS NOT NULL AND ws.created > ? AND ws.created <= ? AND ws.deleted = 0
```
Comment (agorina) [inline migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: I would like to get some input on this: 

For some tables in redshift (master ones) I am including migrate_id, but others do not have it. There is a repo that takes Migrate as an input it uses its id, firstAt and lastAt as parameters to the SQL. To work around it I am including migrate id as '? IS NOT NULL'. Too ugly?

Comment (agorina) [reply migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: Talked offline - will add migrate_id to the staging table to be consistent.

File: migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 2]
```diff
@@ -105,4 +105,51 @@ sql:
               FROM '${archive.root}/${migrate.aws.location}/student_update.part_00000'
               CREDENTIALS 'aws_iam_role=${migrate.aws.redshift.role}'
               FORMAT AS CSV
```
Comment (malaffoon) [inline migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: I know we're controlling the reading/writing of the CSV but what if a student name (or some other field) has a comma in it? Are we escaping things with double quotes?

File: migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 3]
```diff
@@ -105,4 +105,51 @@ sql:
...
+          sql:
+            warehouseRead: >-
+              SELECT
+                ethnicity_id,
+                student_id,
+                ? as migrate_id
+              FROM student_ethnicity
+                JOIN student ws ON ws.id = student_ethnicity.student_id
+              WHERE ws.created > ? AND ws.created <= ? AND ws.deleted = 0
+                INTO OUTFILE S3 '${archive.root}/${migrate.aws.location}/student_ethnicity'
```
Comment (bgsavage) [inline migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: Won't this fail on a second run?  Unless you have something coming behind you to clean up the file.

File: migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml (hunk 2 above)
Comment (agorina) [reply migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: I have it on my 'TODO' list to test this types of things. But thank you for reminding me.

File: migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml (hunk 3 above)
Comment (agorina) [reply migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: Yes, I clean up the file.

Comment (bgsavage) [reply migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: Any potential for a race condition here?

Comment (agorina) [reply migrate-olap/src/main/resources/application.warehouse.to.staging.sql.yml]: Both of our migrates (the reporting and this one) are expected to run as one threaded instance. There are multiple issues with making them scalable. If for some reason two instances are started, they will collide and cause migrate to fail. Once it fails it stays failed until the issues is resolved. This is our crude method of preventing it from being accidently scaled.

**Pull request merged**